assets/js/components/dashboard-sharing/ModuleRecoveryAlert/Error.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * ModuleRecoveryAlert Error component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+
+export default function Error( { recoverableModules, error } ) {
+	return (
+		<div className="googlesitekit-module-recovery-error">
+			{ Object.keys( error ).length === 1 && (
+				<p>
+					{ sprintf(
+						/* translators: %s: Error message */
+						__( 'Error: %s', 'google-site-kit' ),
+						Object.values( error )[ 0 ].message
+					) }
+				</p>
+			) }
+
+			{ Object.keys( error ).length > 1 && (
+				<Fragment>
+					<p>
+						{ __(
+							'The following modules failed to be recovered:',
+							'google-site-kit'
+						) }
+					</p>
+					<ul>
+						{ Object.keys( error ).map( ( module ) => (
+							<li key={ module }>
+								{ recoverableModules[ module ].name }:{ ' ' }
+								{ error[ module ].message }
+							</li>
+						) ) }
+					</ul>
+				</Fragment>
+			) }
+		</div>
+	);
+}
+
+Error.propTypes = {
+	recoverableModules: PropTypes.object,
+	error: PropTypes.object,
```

Comment:
Here the prop type shows `error` is not required, but it is expected to be provided above and `Object.keys` will throw if it does not receive a compatible type.

Secondarily, `error` may not be the appropriate prop here as this object is expected to have a shape which is different than most other errors (e.g. `{ code, message }`). Perhaps `recoveryErrors` may be more appropriate?

Another idea would be to pass only the slugs of the modules which failed and the errors can be selected for each by this component. I'm open to other ideas too, just a thought.

Reply:
I went with `recoveryErrors`. It was a bit tricky to keep track of the modules which failed to recover without storing in a local state, but I was able to accomplish it with a different approach. Please let me know if it looks good. Thank you!